### PR TITLE
Add test cases for JSON resource marshalling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resources Covered :

* PublicKey
* Secret
* Secrets
* EncryptedSecret
* SelectedReposList

Helps with issue : #55

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
